text/0001-dynamic-fees-v1.md

```diff
@@ -0,0 +1,95 @@
+
+Another drawback is the increased debugging difficulty in the event of a failed attestation. In the current system, a mismatched `ResponderId` (prologue) is the most likely source of an `AeadError` during authentication (the other cause would be an active MiTM attack replacing a public key). In the proposed system, because a configured fee is appended to the `ResponderId` to produce the prologue, a mismatch of either value would cause the `AeadError`, and thus both must be checked.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The first alternative considered was simply lowering the fees to 400uMOB, or 0.02USD @ 50USD/MOB. As noted above, while this accomplishes the initial goal of lowering the fees, it does not provide the ability to raise them without another enclave upgrade, which is costly, time-consuming, and will take upwards of 120 days to complete.
+
+The next alternative was using a "fee schedule", or "fee calendar"---that is, rather than having a single fee which is constructed at startup time, use a configuration file which provides a series of "fee ranges" `(fee, startBlock, endBlock)` that are hashed into the key exchange.
+
+This alternative scheme does have the advantage of allowing a reconfiguration to "script" the denial-of-service response, and it's subsequent unwind, in a single restart. The devil, however, is in the details: how should pending and future transactions be handled?
+
+This is actually quite tricky to get right: obviously you want to immediately reject any transactions which do not meet the currently configured fee requirements, but it's entirely possible that your node is already queuing transactions when this comes in.
+
+In this scenario, a client submits a transaction, and it goes into a queue behind hundreds of other transactions. The fee is "correct" for the current block, but the transaction will not actually be entertained for acceptance until after the fee has been raised, and the fee is no longer valid.
+
+Choosing to simply accept that if any node has accepted a proposed transaction at a lower fee, that fee is (or would have been) valid for all nodes is an option, but this does make the ability to enforce fees dependent on the union of all SGX functionality, which is a change from our ad-hoc norms around the use of SGX as a privacy-enhancement technology, rather than a security-critical technology.
+
+The second option is to only check fees lazily, which will produce the scenario where a client could submit a transaction, and have it appear to "time out" later, because it was accepted into the queue at a low fee, then later judged to have an insufficient fee.
```

Comment:
In practice:
1. there should be an expectation in the ecosystem about how much the 'lazy' minimum fee can vary over short timespans
2. the default fee used by clients can be some multiple of the minimum fee, above the potential variance over e.g. 100 blocks (current tombstone limit)

Reply:
actually, both of those points apply to any solution, lazy or otherwise